pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -348,19 +341,26 @@ func (p *provider) Create(machine *v1alpha1.Machine, update cloud.MachineUpdater
 		if err != nil {
 			return nil, fmt.Errorf("failed to create public IP: %v", err)
 		}
+		if !kuberneteshelper.HasFinalizer(machine, finalizerPublicIP) {
+			if machine, err = update(machine.Name, func(updatedMachine *v1alpha1.Machine) {
+				updatedMachine.Finalizers = append(updatedMachine.Finalizers, finalizerPublicIP)
+			}); err != nil {
+				return nil, err
+			}
+		}
 	}
 
+	iface, err := createNetworkInterface(context.TODO(), ifaceName, machine.UID, config, publicIP)
```

Comment:
Sidequestion:
What happens if the `Create` fuctions dies after creating a network interface?
Will it reuse the already created interface on the next sync? Or will it fail with a `AlreadyExists` err?

Reply:
I _think_ it will be a noop, because the underlying func from the Azure SDK is called `CreateOrUpdate`

Reply:
Perfect!